Webhook documentation (#35)
sieve-data/docs: docs/guide/custom-apps/using-webhooks.mdx

---
title: "Using Webhooks"
description: "Webhooks provide a way for notifications to be delivered to an external web server when a Sieve job completes."
---

## **About webhooks**

> As alternative to polling the Sieve API for a job to complete, webhooks allow Sieve to notify your API when results are ready. This eliminates unnecessary requests made by polling, and can get results back faster. In addition, webhooks can be triggered by via the Sieve website, which can enable complex workflows with an accessible interface.

Reviewer: ```suggestion
As alternative to polling the Sieve API for a job to complete, webhooks allow Sieve to notify your API when results are ready. This eliminates unnecessary requests made by polling, and can get results back faster. In addition, webhooks can be triggered by the Sieve website, which can enable complex workflows with an accessible interface.
```

Submit a new job by sending a POST request to the **`/v2/push`** endpoint:

```python
  curl 'https://mango.sievedata.com/v2/push' \
  -H 'X-API-Key: <your api key>' \
  --data-raw '{
    "function": "sieve/speech_transcriber",
    "inputs": {
      "file": {
        "url": "https://storage.googleapis.com/sieve-public-data/assets/dub.m4a"
     },
    "webhooks": [{
            "type": "<event type>",
            "url": "<your webhook url>"
        }]
    }
  }'
```

Once the job completes, you'll receive a notification on [webhook.site](https://webhook.site/) with the output of the job. The response output is in the following format:

## **Using webhooks via the Python client**

Webhooks can also be set via the Python client with the `webhooks` parameter.

```python
f = sieve.function.get("...")
f.run(
    target_audio,
    webhooks=[
        {
            "type": "<event type>",
            "url": "<your webhook url>"
        }
    ]
)
```

The code will run as usual, with the added benefit of a webhook being triggered once the job is complete.

> ### **Using webhooks via the Sieve dashboard**

Reviewer: <img width="291" alt="image" src="https://github.com/sieve-data/docs/assets/6136843/378f90c2-9b2e-44a8-85da-14031edd69f8">
wrong hierarchy, this should fix

```suggestion
## **Using webhooks via the Sieve dashboard**
```

> Webhook can also be set on a per-function basis via. the dashboard. In other words, every time the model is run the webhook will be triggered, including via. the REST API. Add the webhook on the Settings page once you have an app selected.

Reviewer: ```suggestion
Webhooks can also be set on a per-function basis via the dashboard. In other words, every time the function is run (from the UI, REST API, or Python client), the webhook will be triggered. Add the webhook on the Settings page once you have a function selected.
```

> 

Reviewer: are we adding an image here?

---
title: "Using Webhooks"
description: "Webhooks provide a way for notifications to be delivered to an external web server when a Sieve job completes."
---

## **About webhooks**

As alternative to polling the Sieve API for a job to complete, webhooks allow Sieve to notify your API when results are ready. This eliminates unnecessary requests made by polling, and can get results back faster. In addition, webhooks can be triggered by the Sieve website, which can enable complex workflows with an accessible interface.

## **Event types**

Webhooks can be triggered by the following event types:

- `job_complete` - triggered when a job completes successfully or with an error.
- `job_started` - triggered when a job has exited the queue and is being processed.

> More event types will be added in the future. See the [API reference](/reference-v2/api/webhook) for more information.

Reviewer: ```suggestion
More event types will be added in the future. See the [API reference](/reference-v2/api/webhook) for more response information.
```
